flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamNode.java

```diff
@@ -336,15 +341,30 @@ public void setUserHash(String userHash) {
 		this.userHash = userHash;
 	}
 
-	@VisibleForTesting
 	public void setSortedInputs(boolean sortedInputs) {
 		this.sortedInputs = sortedInputs;
 	}
 
-	boolean getSortedInputs() {
+	public boolean getSortedInputs() {
 		return sortedInputs;
 	}
 
+	public void setStateBackend(StateBackend stateBackend) {
```

Comment:
From what I understand, the flow is that we set the batch `StateBackend` and the `timerService` in the translator to the `StreamNode` so that the `StreamJobGraphGenerator` can pick it up. Why not setting the state backend and the timer service at the `StreamGraph` level (e.g. in. the `StreamGraphGenerator.configureStreamGraph()`) from where the `StreamJobGraphGenerator` can pick it up.

This seems to be able to reduce the changes in the `StreamNode` and the `StreamJobGraphGenerator`. 

WDYT @dawidwys ?

Reply:
After offline sync and some more thinking and I thought it is actually dangerous to set the state backend on a per node level. I will revert back to setting state backend only on the StreamGraph level and throw an exception if there is an InputSelectable operator for now.